### ISSUE TITLE
Adding a docker image

### ISSUE THREAD

**Issue by Mattie112** [AGuyNamedJens/FactorioChatBot#3]
Hi,

I'd like to run this in a docker image (to allow easy deployment). Would you accept a PR adding this? Then I can look into it?

And just to understand:

RCON is needed on order to send messages from discord -> factorio?
And you use the log file to go from factorion -> discord? Or where is that used for?

**Comment by AGuyNamedJens** [comment]
Hi,

Sure thing, will look into that feature soon.

RCON is indeed used to send messages to the factorio server, and the log file is used to read the chat from the game, since the only other way would be to use a custom scenario, which disables achievements by default which is exactly not how this bot is supposed to work.

**Comment by Mattie112** [comment]
Yeah sounds reasonable. I thought chats would also be supplied by RCON but it does not.

Do you have docker experience? A dockerfile like this should work

```
FROM node:16
WORKDIR /opt/project
COPY index.js /opt/project/index.js
ADD package-lock.json /opt/project/package-lock.json
ADD package.json /opt/project/package.json
RUN cd /opt/project
RUN npm install

# And then move the config.json to this folder
VOLUME /opt/config
# To bind the log fine (that may be generated by an other container)
VOLUME /opt/factorio-log

CMD node .
```

Or you can skip the config dir if you only use environment variables.

You can build it with `docker build -t versiontag .` and then try it with `docker run --rm -it versiontag`

(at least this is how I run it now)

edit:
I'm happy to submit a PR if you want

**Comment by AGuyNamedJens** [comment]
Yeah, for some reason RCON doesn't provide outputs unless you have a custom scenario that makes RCON do it.

I do have docker experience, I run my own Discord bot (not FactorioChatBot) under docker myself.
I will probably move config.json to a .env file in the future as well. But that dockerfile could work.

I'll have to make some time free to start working on the new updates. feel free to make a PR.

**Comment by AGuyNamedJens** [comment]
I've went ahead and added a dockerfile, will test it later as docker is currently not installed on my system.

I'll start doing more changes once I have more time, such as adding it into the wiki.

**Comment by Mattie112** [comment]
Thanks! I'll look into it tonight!

**Comment by Mattie112** [comment]
https://github.com/AGuyNamedJens/FactorioChatBot/pull/7

If you want to test it I have it pushed to my DockerHub: https://hub.docker.com/repository/docker/mattie112/factorio-chat-bot
Currently running great on my Unraid machine :)

**Comment by AGuyNamedJens** [comment]
Sadly Discord is currently having an outage, however besides for my comment on the new config location it looks good to me.
